src/iter.rs

```diff
@@ -93,24 +93,26 @@ impl<B: 'static> IterNames<B> {
     /// check whether or not there are any bits that didn't correspond
     /// to a valid flag remaining.
     pub fn remaining(&self) -> &B {
-        &self.state
+        &self.remaining
     }
 }
 
 impl<B: Flags> Iterator for IterNames<B> {
     type Item = (&'static str, B);
-    
+
     fn next(&mut self) -> Option<Self::Item> {
         while let Some(flag) = self.flags.get(self.idx) {
             // Short-circuit if our state is empty
-            if self.state.is_empty() {
+            if self.remaining.is_empty() {
                 return None;
             }
 
             self.idx += 1;
 
             let bits = flag.value().bits();
 
+            // TODO: Try another state value; `|` the bits into it, and don't yield unless they change the value of the output; this would avoid composite flags
```

Comment:
Don’t need this: we can just xor the incoming flag against our state and if it’s non-zero then it represents a “new” flag we haven’t yielded yet.